backend/lib/lotus/transform.js

```
@@ -0,0 +1,39 @@
+import { readFile } from 'node:fs/promises'
+import { fromDSL } from '@ipld/schema/from-dsl.js'
+import { create } from '@ipld/schema/typed.js'
+
+class Transformer {
+  // A transformer which takes in a json object and returns a typed ClaimEvent object
+  #claimEventTransformer
+
+  async build () {
+    const schemaPath = new URL('../lotus/builtin-actor-events-schemas.ipldsch', import.meta.url)
+    // TODO: Catch and log errors
```

If you don't mind, let's ignore error handling for now and focus on getting a full vertical slice completed first, including the loop observing the events via the RPC API and updating the database.

Then we can clean up the rough edges and make the implementation more robust.

Leaving TODO comments like this one is a great solution for this 👍🏻

---

Yes, I was thinking of adding rigorous logging for the server at some point. But implementing logging when the server itselve is not yet up seemed a bit premature for me. I just wanted to give myselve a reminder where possible errors could spawn.